services-core/src/economic_viability.rs

```diff
@@ -187,13 +190,41 @@ impl FixedEconomicViabilityComputer {
 #[async_trait::async_trait]
 impl EconomicViabilityComputing for FixedEconomicViabilityComputer {
     async fn min_average_fee(&self) -> Result<u128> {
-        self.min_average_fee
-            .ok_or_else(|| anyhow!("no min average fee set"))
+        Ok(self.min_average_fee)
     }
 
     async fn max_gas_price(&self, _: EconomicViabilityInfo) -> Result<U256> {
-        self.max_gas_price
-            .ok_or_else(|| anyhow!("no max gas price set"))
+        Ok(self.max_gas_price)
+    }
+}
+
+/// Use whichever computer yields a more trader friendly min average fee.
+pub struct CombinedEconomicViabilityComputer {
+    fixed: FixedEconomicViabilityComputer,
+    // This is always DynamicViabilityComputing but for easier mock testing we use box.
+    dynamic: Box<dyn EconomicViabilityComputing>,
+}
+
+#[async_trait::async_trait]
+impl EconomicViabilityComputing for CombinedEconomicViabilityComputer {
+    async fn min_average_fee(&self) -> Result<u128> {
+        let dynamic = self.dynamic.min_average_fee().await?;
```

Comment:
Do we really want to fail if the dynamic fails? Why not just fallback to fixed?

Reply:
That makes sense. Changed it to warn log and fallback.